src/TokenChecks.sol

```diff
@@ -0,0 +1,565 @@
+        uint256 prevRecipientBalance = TokenLike(_token).balanceOf(address(0xBEEF));
+        uint256 prevSenderBalance = TokenLike(_token).balanceOf(address(this));
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(address(this), address(0xBEEF), 1e18);
+        assertTrue(TokenLike(_token).transfer(address(0xBEEF), 1e18));
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply);
+        assertEq(TokenLike(_token).balanceOf(address(this)), prevSenderBalance - 1e18);
+        assertEq(TokenLike(_token).balanceOf(address(0xBEEF)), prevRecipientBalance + 1e18);
+    }
+
+    function checkTransferFrom(address _token) public {
+        address from = address(0xABCD);
+        deal(_token, from, TokenLike(_token).balanceOf(address(this)) + 1e18, true);
+        vm.prank(from);
+        TokenLike(_token).approve(address(this), 1e18);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevRecipientBalance = TokenLike(_token).balanceOf(address(0xBEEF));
+        uint256 prevFromBalance = TokenLike(_token).balanceOf(from);
+        uint256 prevAllowance = TokenLike(_token).allowance(from, address(this));
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(from, address(0xBEEF), 1e18);
+        assertTrue(TokenLike(_token).transferFrom(from, address(0xBEEF), 1e18));
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply);
+        assertEq(TokenLike(_token).allowance(from, address(this)), prevAllowance - 1e18);
+        assertEq(TokenLike(_token).balanceOf(from), prevFromBalance - 1e18);
+        assertEq(TokenLike(_token).balanceOf(address(0xBEEF)), prevRecipientBalance + 1e18);
+    }
+
+    function checkInfiniteApproveTransferFrom(address _token) public {
+        address from = address(0xABCD);
+        deal(_token, from, TokenLike(_token).balanceOf(address(this)) + 1e18, true);
+        vm.prank(from);
+        vm.expectEmit(true, true, true, true);
+        emit Approval(from, address(this), type(uint256).max);
+        TokenLike(_token).approve(address(this), type(uint256).max);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevRecipientBalance = TokenLike(_token).balanceOf(address(0xBEEF));
+        uint256 prevFromBalance = TokenLike(_token).balanceOf(from);
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(from, address(0xBEEF), 1e18);
+        assertTrue(TokenLike(_token).transferFrom(from, address(0xBEEF), 1e18));
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply);
+        assertEq(TokenLike(_token).allowance(from, address(this)), type(uint256).max);
+        assertEq(TokenLike(_token).balanceOf(from), prevFromBalance - 1e18);
+        assertEq(TokenLike(_token).balanceOf(address(0xBEEF)), prevRecipientBalance + 1e18);
+    }
+
+    function checkDecreaseAllowanceInsufficientBalance(address _token, string memory _contractName) public {
```

Comment:
I think this test should be called better called: `checkDecreaseAllowanceInsufficientAllowance` and be moved right after the regular one.

Reply:
Good catch